resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -289,4 +278,90 @@ public String toString() {
             }
         }
     }
+
+    private static final class Entries {
+        private final ReadWriteLock lock = new ReentrantReadWriteLock();
+        private final Lock readLock = lock.readLock();
+        private final Lock writeLock = lock.writeLock();
+
+        // Access to this field is guarded by the ReadWriteLock.
+        //
+        // It's important that this List will never be modified itself but just replaced when an new entry is added,
+        // or removed.
+        private List<DefaultDnsCacheEntry> entries;
+
+        Entries(DefaultDnsCacheEntry entry) {
+            entries = Collections.unmodifiableList(Collections.singletonList(entry));
+        }
+
+        List<DefaultDnsCacheEntry> get() {
+            readLock.lock();
+            try {
+                return entries;
+            } finally {
+                readLock.unlock();
```

Comment:
the list escapes the lock here which may violate what ever locking/synchronization enforced by this class internally.